src/packet_analysis/protocol/teredo/CMakeLists.txt

```diff
@@ -1,7 +1,10 @@
-include(ZeekPlugin)
-
-zeek_plugin_begin(Zeek Teredo)
-zeek_plugin_cc(Teredo.cc Plugin.cc)
-zeek_plugin_bif(events.bif)
-zeek_plugin_bif(functions.bif)
-zeek_plugin_end()
+zeek_add_plugin(
+    Zeek
```

Comment:
This is `Zeek` in the existing code, but probably should be made `PacketAnalyzer` like the others.

Reply:
Done.

Reply:
Actually, things break when moving this to `Zeek`. It's in the Zeek namespace on current master: https://github.com/zeek/zeek/blob/master/src/packet_analysis/protocol/teredo/CMakeLists.txt#L3.

There's also some weirdness with some BIFs being baked directly into Zeek. The Broker BIFs for example have to use the "alternative mode" to get built directly into Zeek or else the `btests` complain. Maybe the tests are just overly strict? If we do clean that up, I'd prefer doing that as a followup, though, to focus this PR on modernizing the CMake without changing the behavior (wherever possible).

Reply:
Yeah, this is weird. For some reason it doesn't like the definitions in functions.bif in both the gtpv1 and teredo analyzers if you move them to the other namespace.

Reply:
Ah, I had to nuke build/scripts/base/bif/plugins/Zeek_Teredo.functions.bif.zeek and then fix scripts/base/packet-protocols/teredo/main.zeek to point at the right file. I agree this is out of scope for these changes. I'm fine with making a note somewhere and coming back to them later.